LeGO-LOAM/src/featureAssociation.cpp

```diff
@@ -180,13 +180,37 @@ class FeatureAssociation{
     cv::Mat matP;
 
     int frameCount;
+    int imuQueLength;
+    float surfThreshold;
+    int nearestFeatureSearchSqDist;
+    float edgeThreshold;
+    int N_SCAN ;
+    int Horizon_SCAN ;
+    float ang_res_x ;
+    float ang_res_y ;
+    float ang_bottom ;
+    int groundScanInd ;
 
 public:
 
     FeatureAssociation():
         nh("~")
         {
-
+        nh.getParam("scanPeriod", scanPeriod);
+        string imuTopic;
```

Comment:
For this you could use topic remapping in the launchfiles. That way you make the algorithm more flexible to ROS communication layer changes, without needing to recompile.